src/v/archival/archiver_operations_impl.cc

```diff
@@ -171,23 +198,252 @@ class archiver_operations_impl : public archiver_operations_api {
     /// the uploaded manifest. The state of the uploaded manifest doesn't
     /// include uploaded segments because they're not admitted yet.
     ss::future<result<upload_results_list>> schedule_uploads(
-      retry_chain_node&,
-      reconciled_upload_candidates_list,
-      bool) noexcept override {
-        co_return error_outcome::unexpected_failure;
+      retry_chain_node& workflow_rtc,
+      reconciled_upload_candidates_list bundle,
+      bool inline_manifest_upl) noexcept override {
+        try {
+            upload_results_list result;
+            auto partition = _pm->get_partition(bundle.ntp);
+            if (partition == nullptr) {
+                // maybe race condition (partition was stopped or moved)
+                vlog(
+                  _rtclog.debug,
+                  "schedule_uploads - can't find partition {}",
+                  bundle.ntp);
+                co_return error_outcome::unexpected_failure;
+            }
+            std::deque<cloud_storage::segment_meta> metadata;
+            chunked_vector<ss::future<aggregated_upload_result>> uploads;
+            for (auto& upl : bundle.results) {
+                metadata.push_back(upl->metadata);
+                uploads.emplace_back(
+                  upload_candidate(workflow_rtc, partition, upl));
+            }
+            model::offset projected_clean_offset
+              = partition->manifest().get_insync_offset();
+            if (inline_manifest_upl) {
+                const auto& manifest = partition->manifest();
+                const auto estimated_manifest_upl_size
+                  = manifest.estimate_serialized_size();
+                auto key = partition->get_remote_manifest_path(manifest);
+                uploads.emplace_back(
+                  _api
+                    ->upload_manifest(
+                      _bucket, manifest, std::move(key), workflow_rtc)
+                    .then([estimated_manifest_upl_size](
+                            cloud_storage::upload_result r) {
+                        return aggregated_upload_result{
+                          .code = r,
+                          .put_requests = 1,
+                          // Use size estimate instead of the actual value
+                          .bytes_sent = estimated_manifest_upl_size};
+                    }));
+            }
+
+            // Wait for all uploads to complete
+            auto result_vec = co_await ss::when_all(
+              uploads.begin(), uploads.end());
+            size_t num_put_requests = 0;
+            size_t num_bytes_sent = 0;
+
+            // Process manifest upload results.
+            model::offset manifest_clean_offset;
+            if (inline_manifest_upl) {
+                if (result_vec.back().failed()) {
```

Comment:
nit: is there a more reliable way to identify the manifest upload out of `result_vec`? Otherwise it looks like with some future code change (eg another upload is added after the manifest) we might end up making the following decisions  on the basis of a non-manifest upload.